### ISSUE TITLE
Add libsystemd-dev for swift-systemd

### ISSUE THREAD

**Issue by xtremekforever** [SwiftPackageIndex/SwiftPackageIndex-Server#3366]
**Which package(s) are showing incorrect data?**

- https://swiftpackageindex.com/xtremekforever/swift-systemd

**Please explain what data is incorrect and what you expected to see**

The `swift-systemd` package requires the systemd headers (`libsystemd-dev`) to be available in Linux for compilation, but it is not included in the default SPI builder image for Linux.

**If the problem relates to the compatibility matrix, does it apply to specific builds?**

Please find the builds exhibiting the problem from the "Full Package Compatibility Report" menu item in the side menu and link the build reports here.

- 6.0: https://swiftpackageindex.com/builds/BE0BF622-E234-464E-AB65-27E9AA6CC299
- 5.10: https://swiftpackageindex.com/builds/F2682245-57EE-4F91-A1A1-AF9D383495E5
- 5.9: https://swiftpackageindex.com/builds/349A44D2-7473-45CA-9B40-2605B8FA47D0
- 5.8: https://swiftpackageindex.com/builds/007C98D0-FBE1-4C01-990D-566BF3E062D6

Where possible, try and replicate the problem you have found by running the "Build Command" at the top of a build report. This command is what our build servers execute.

**Screenshots**

```bash
[110/165] Compiling Systemd SystemdNotifier.swift
<module-includes>:1:10: note: in file included from <module-includes>:1:
#include "systemd.h"
         ^
/host/spi-builder-workspace/Sources/CSystemd/systemd.h:1:10: error: 'systemd/sd-daemon.h' file not found
#include <systemd/sd-daemon.h>
         ^
/host/spi-builder-workspace/Sources/Systemd/SystemdHelpers.swift:4:8: error: could not build C module 'CSystemd'
import CSystemd
       ^
[111/165] Compiling Systemd SystemdHelpers.swift
<module-includes>:1:10: note: in file included from <module-includes>:1:
#include "systemd.h"
         ^
/host/spi-builder-workspace/Sources/CSystemd/systemd.h:1:10: error: 'systemd/sd-daemon.h' file not found
#include <systemd/sd-daemon.h>
         ^
/host/spi-builder-workspace/Sources/Systemd/SystemdHelpers.swift:4:8: error: could not build C module 'CSystemd'
import CSystemd
       ^
```

**Additional context**

The package README.md shows how to install `libsystemd-dev` in Ubuntu or other hosts. This is a simple dependency which may be useful to include as a general dependency, but so far it's only really used for this package so I'd understand if it would require a special image.

**Comment by finestructure** [comment]
That should be simple enough to add to the base image, thanks for the report!

**Comment by finestructure** [comment]
It's now passing on `main`. `0.0.5` has what seems like a genuine build error:

```
/host/spi-builder-workspace/Sources/Example/Example.swift:20:42: error: argument passed to call that takes no arguments
18 |     configuration: .init(
19 |         services: [
20 |             .init(service: SystemdService(watchdogEnabled: true, watchdogInterval: .seconds(10)))
   |                                          `- error: argument passed to call that takes no arguments
```

![CleanShot 2024-09-09 at 10 39 13@2x](https://github.com/user-attachments/assets/8b947fcf-64f0-42e8-a2a3-30887050d8d6)

**Comment by xtremekforever** [comment]
Yep, that was just a typo in the example project that was included in 0.0.5. I'll be tagging a new version soon with a couple of cleanups and documentation. Thanks for getting this added!!
